packages/lib/gpc/src/gpcChecks.ts

```diff
@@ -413,25 +459,48 @@ export function checkProofInputsForConfig(
       }
 
       // If this entry identifies the owner, we should have a matching Identity.
-      if (entryConfig.isOwnerID) {
+      if (entryConfig.isOwnerID !== undefined) {
         hasOwnerEntry = true;
-        if (proofInputs.owner === undefined) {
+        if (
+          proofInputs.owner?.semaphoreV3 === undefined &&
+          proofInputs.owner?.semaphoreV4 === undefined
+        ) {
           throw new Error(
             "Proof configuration expects owner, but no owner identity given."
           );
         }
-        if (podValue.value !== proofInputs.owner.semaphoreV3.commitment) {
-          throw new Error(
-            `Configured owner commitment in POD doesn't match given Identity.`
-          );
-        }
 
-        // Owner commitment value must be a cryptographic number, so that its
-        // plaintext value can be included in the circuit.
-        if (podValue.type !== "cryptographic") {
-          throw new Error(
-            "Owner identity commitment must be of cryptographic type."
-          );
+        for (const [ownerIDType, ownerCommitment] of [
```

Comment:
Maybe use `ownerID` rather than `ownerCommitment` now that it's not the same thing as a semaphore commitment?